### PR TITLE
[constraint] adds JointConstraint

### PR DESCRIPTION
**Pull request by EulalieCoevoet**
Thank you @TanguyNav for this work!

Proposition of changes:

- Instead of `maxPositiveDisp` / `maxNegativeDisp`, I would prefer using `maxDisplacement` / `minDisplacement`. When we want a positive minimum it's really weird to give a negative value. Even in the general case I find it confusing (I don't like it either in CableConstraint...). 
- For me the references to both displacement and angle are confusing. It's true that the joint does not have to be angular and could be a linear displacement. I would remove all the references to `angle`.
- From a user point of view (from the python script) we use the data names `value` and `valueType` in the other constraints (CableConstraint, SurfacePressureConstraint, PositionConstraint). I would prefer to keep it uniform and thus avoid using `imposedValue` instead of `value`.

### CODE REVIEW

**Review by alxbilger — CHANGES_REQUESTED** [review]
It would be great to have an example scene.

**Comment by EulalieCoevoet** [comment]
Well if no one disagree I'll make the changes I proposed.

**Comment by EulalieCoevoet** [comment]
This is good for me. I'll merge it in two weeks, unless you need more time to look at it @TanguyNav @alxbilger ?

**Review by alxbilger — APPROVED** [review]

File: examples/component/constraint/JointConstraint/JointConstraint.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import os
+from math import pi
+
+dirPath = os.path.dirname(os.path.abspath(__file__)) + '/'
+
+
+def createScene(rootNode):
+    rootNode.addObject('RequiredPlugin', name='SoftRobots')
+    rootNode.addObject('RequiredPlugin', name='SoftRobots.Inverse')
```
Comment (alxbilger) [inline examples/component/constraint/JointConstraint/JointConstraint.py:9]: @EulalieCoevoet Could you make sure that this plugin is not required? Thanks

Comment (EulalieCoevoet) [reply examples/component/constraint/JointConstraint/JointConstraint.py:9]: Oups sorry. Not needed indeed.

**Pull request merged**